### PR TITLE
Update naming of methods for Coulomb logarithms

### PR DESCRIPTION
**Pull request by Tiger-Du**
- [x] I have added a changelog entry for this pull request.
- [x] If adding new functionality, I have added tests and
      docstrings.
- [ ] I have fixed any newly failing tests.

-| Part of Issue #931.
-| Renames the methods for Coulomb logarithms to "LS", "LS_min_I", "LS_full_I", "LS_clamp", "hyp_LS", "hyp_LS_max_I", and "hyp_LS_full_I". Underscores could be removed, "hyp" could be shortened to "H", and "I" could be changed to "i", "interp", or something similar.
-| Other minor updates.

### CODE REVIEW

File: plasmapy/formulary/collisions.py
Comment (namurphy) [inline plasmapy/formulary/collisions.py:89]: For this one, I'm wondering if `"classical"` would be better.

File: plasmapy/formulary/collisions.py
Comment (namurphy) [inline plasmapy/formulary/collisions.py:229]:
```suggestion
        :math:`\Lambda < 0`, therefore a clamp is placed at Lambda_min = 2.
    hyp_LS ("GMS-4")
        4th method listed in Table 1 of reference [4].
        Spitzer-like extension to Coulomb logarithm by noting that
        Coulomb collisions take hyperbolic trajectories. Removes
        divergence for small ``bmin`` issue in classical Landau-Spitzer
        approach, so ``bmin`` can be zero. Also doesn't break down as
        :math`\Lambda < 0` is now impossible, even when coupling parameter is large.
```
Some minor formatting suggestions...

File: plasmapy/formulary/collisions.py
Comment (namurphy) [inline plasmapy/formulary/collisions.py:228]:
```suggestion
        divergence for small ``bmin`` issue in classical Landau-Spitzer
        approach, so ``bmin`` can be zero. Also doesn't break down as
```

File: plasmapy/formulary/collisions.py
Comment (namurphy) [inline plasmapy/formulary/collisions.py:234]:
```suggestion
        Similar to GMS-4, but setting ``bmin`` as distance of closest approach
        and ``bmax`` interpolated between Debye length and ion sphere radius.
        :math:`\Lambda < 0` impossible.  5th method listed in Table 1 of reference [4].
```
I'm wondering if it would be helpful to move the sentences like "5th method list in..." to the end of each method description.

**Review by namurphy — CHANGES_REQUESTED** [review]
Thank you for working on this!  Right now I'm mostly wondering about the balance between abbreviation and readability.  The more abbreviated the method names, the harder they are to read and understand without going into the documentation.  It'll be necessary to update the [tests for `Coulomb logarithm`](https://github.com/PlasmaPy/PlasmaPy/blob/b42ad32f26aaf9cf1bf716ed069f3ac0f7eefe42/plasmapy/formulary/tests/test_collisions.py#L28) too.  Thank you again!

**Comment by Tiger-Du** [comment]
From helpful discussion with @namurphy, @svincena, and @rocco8773, I think the following should be done.

- [ ] Each method should retain its original name for now, and the proposed names should be made aliases. In the future (perhaps before this pull request is done), we may decide to break backward compatibility by removing the "GMS" names.

- [ ] The documentation should be expanded to include the formula for each method, a more direct explanation of the use cases of each one, and a more direct explanation overall of PlasmaPy's stance on these methods.

- [ ] A Jupyter Notebook should be made in a new pull request to illustrate the differences among these methods.

**Comment by lemmatum** [comment]
Thanks for cleaning this up!

**Comment by Tiger-Du** [comment]
@lemmatum - No problem! I'm happy to help.

**Comment by Tiger-Du** [comment]
@namurphy - Thank you for your review. All of the changes you suggested are implemented, except for the backticks around bmin and bmax in the docstring. I think it is best to use LaTeX formatting  because the programming variables, `bmin` and `bmax`, are not parameters that the user interacts with in `Coulomb_logarithm` but are instead local variables of this function. In addition, I added the LaTeX-formatted formulas of each of the methods to the docstring, so I think that formatting bmin and bmax in the docstring as mathematical variables rather than programming variables would improve readability.

**Comment by Tiger-Du** [comment]
Before I update the tests and the functions that call `Coulomb_logarithm`, we should decide whether we would like to go forward with this new naming scheme or whether we should make any changes to it. @namurphy - Does this naming scheme look alright to you?

It may be helpful to look at the version of the docs that was built for this pull request ([link](https://plasmapy--962.org.readthedocs.build/en/962/api/plasmapy.formulary.collisions.Coulomb_logarithm.html)) in comparison to the stable version ([link](https://docs.plasmapy.org/en/stable/api/plasmapy.formulary.collisions.Coulomb_logarithm.html#plasmapy.formulary.collisions.Coulomb_logarithm)).

Last, I think we should decide answers to these questions:
1) If Option 7, `"hlsfullinterp"`, is the best method and our recommended method, should it be the default method?
2) Should we add `bmin` and `bmax` as parameters of `Coulomb_logarithm` in case a user already knows them? We could perhaps make it so that a user only needs to specify `bmin` and `bmax` (and optionally `method`) to compute the Coulomb logarithm in case the user does not want to specify the other parameters or does not know the values of the other parameters. Is this something that would be useful for a typical plasma physicist?

**Review by namurphy — CHANGES_REQUESTED** [review]
Once again, thank you for working on this!  Coulomb logarithms turned out to be a lot more complicated than I had originally thought when we first worked on this function.

- My preference at the moment would be to have the abbreviations be `"LS"` for classical and  `"GMS-1"` through `"GMS-6"` for the rest because that's a lot easier to remember.  I'm also thinking that if the function will still accept `"GMS-1"` through `"GMS-6"`, it'd be better to keep that in the documentation.
- I think I'd also find it more readable to have underscores like you originally did, like `"ls_full_interp"`.  My suggestions at the moment are:
```
    2. `"ls_min_interp"` or `"GMS-1"`
    3. `"ls_full_interp"` or `"GMS-2"`
    4. `"ls_clamp_mininterp"` or `"GMS-3"`
    5. `"hls_min_interp"` or `"GMS-4"`
    6. `"hls_max_interp"` or `"GMS-5"`
    7. `"hls_full_interp"` or `"GMS-6"`
```
Thank you again!

File: plasmapy/formulary/collisions.py
Comment (Tiger-Du) [reply plasmapy/formulary/collisions.py:156]: Yes, I noticed that colons are used in this way in the docstring of `Alfven_speed` when I compared it with the docstring of `Coulomb_logarithm` for reference. Without colons, the warnings are displayed in the docs as literal text, "~astropy.units.UnitsWarning" and "~plasmapy.utils.RelativityWarning", rather than as code that is hyperlinked to the documentation of the warnings. I didn't find any information online on why the colons work. Maybe @StanczakDominik or @rocco8773 knows?

File: plasmapy/formulary/collisions.py
Comment (namurphy) [inline plasmapy/formulary/collisions.py:162]:
```suggestion
    **Overview of Supported Methods of Computing the Coulomb Logarithm**
```
Possible slight wording change

File: plasmapy/formulary/collisions.py
Comment (namurphy) [inline plasmapy/formulary/collisions.py:690]:
```suggestion
    all GMS methods, except for GMS-5, where ``b_min`` is simply set to
```
Minor restructuredText formatting change.

**Review by namurphy — COMMENTED** [review]
Just took a quick look at it, and it's looking pretty close to ready.  I had a few small suggested changes.  I'll take a closer look at it this week but I don't expect to suggest anything major.  Thank you again!

File: plasmapy/formulary/collisions.py
Comment (namurphy) [reply plasmapy/formulary/collisions.py:156]: Interesting, thanks!

File: plasmapy/formulary/collisions.py
Comment (namurphy) [inline plasmapy/formulary/collisions.py:199]:
```suggestion
    In this section, further information about each method, such as about
    interpolation and other special features, is documented. Please refer
    to Reference [4]_ for additional information about these methods.
```
Inconsequential formatting suggestion for line length

File: plasmapy/formulary/collisions.py
Comment (namurphy) [inline plasmapy/formulary/collisions.py:642]:
```suggestion
        (``"classical"`` or ``"LS"``). The other 6 supported methods are 
        ``"ls_min_interp"``, ``"ls_full_interp"``, ``"ls_clamp_mininterp"``, 
        ``"hls_min_interp"``, ``"hls_max_interp"``, and ``"hls_full_interp"``.
        Please refer to the docstring of `Coulomb_logarithm` for more
        information about these methods.
```
Another minor change for line length

File: plasmapy/formulary/collisions.py
Comment (namurphy) [inline plasmapy/formulary/collisions.py:1066]:
```suggestion
        The default method is the classical straight-line Landau-Spitzer
        method (``"classical"`` or ``"LS"``). The other 6 supported methods
        are ``"ls_min_interp"``, ``"ls_full_interp"``, ``"ls_clamp_mininterp"``,
        ``"hls_min_interp"``, ``"hls_max_interp"``, and ``"hls_full_interp"``.
        Please refer to the docstring of `Coulomb_logarithm` for more
        information about these methods.
```

File: plasmapy/formulary/collisions.py
Comment (namurphy) [inline plasmapy/formulary/collisions.py:1195]:
```suggestion
        The default method is the classical straight-line Landau-Spitzer
        method (``"classical"`` or ``"LS"``). The other 6 supported methods
        are ``"ls_min_interp"``, ``"ls_full_interp"``, ``"ls_clamp_mininterp"``,
        ``"hls_min_interp"``, ``"hls_max_interp"``, and ``"hls_full_interp"``.
        Please refer to the docstring of `Coulomb_logarithm` for more
        information about these methods.
```

File: plasmapy/formulary/collisions.py
Comment (namurphy) [inline plasmapy/formulary/collisions.py:1335]:
```suggestion
        The default method is the classical straight-line Landau-Spitzer
        method (``"classical"`` or ``"LS"``). The other 6 supported methods
        are ``"ls_min_interp"``, ``"ls_full_interp"``, ``"ls_clamp_mininterp"``,
        ``"hls_min_interp"``, ``"hls_max_interp"``, and ``"hls_full_interp"``.
        Please refer to the docstring of `Coulomb_logarithm` for more
        information about these methods.
```

File: plasmapy/formulary/collisions.py
Comment (namurphy) [inline plasmapy/formulary/collisions.py:1456]:
```suggestion
        The default method is the classical straight-line Landau-Spitzer
        method (``"classical"`` or ``"LS"``). The other 6 supported methods
        are ``"ls_min_interp"``, ``"ls_full_interp"``, ``"ls_clamp_mininterp"``,
        ``"hls_min_interp"``, ``"hls_max_interp"``, and ``"hls_full_interp"``.
        Please refer to the docstring of `Coulomb_logarithm` for more
        information about these methods.
```

File: plasmapy/formulary/collisions.py
Comment (namurphy) [inline plasmapy/formulary/collisions.py:1586]:
```suggestion
        The default method is the classical straight-line Landau-Spitzer
        method (``"classical"`` or ``"LS"``). The other 6 supported methods
        are ``"ls_min_interp"``, ``"ls_full_interp"``, ``"ls_clamp_mininterp"``,
        ``"hls_min_interp"``, ``"hls_max_interp"``, and ``"hls_full_interp"``.
        Please refer to the docstring of `Coulomb_logarithm` for more
        information about these methods.
```

File: plasmapy/formulary/collisions.py
Comment (namurphy) [inline plasmapy/formulary/collisions.py:1716]:
```suggestion
        The default method is the classical straight-line Landau-Spitzer
        method (``"classical"`` or ``"LS"``). The other 6 supported methods
        are ``"ls_min_interp"``, ``"ls_full_interp"``, ``"ls_clamp_mininterp"``,
        ``"hls_min_interp"``, ``"hls_max_interp"``, and ``"hls_full_interp"``.
        Please refer to the docstring of `Coulomb_logarithm` for more
        information about these methods.
```

File: plasmapy/formulary/collisions.py
Comment (namurphy) [inline plasmapy/formulary/collisions.py:1837]:
```suggestion
        The default method is the classical straight-line Landau-Spitzer
        method (``"classical"`` or ``"LS"``). The other 6 supported methods
        are ``"ls_min_interp"``, ``"ls_full_interp"``, ``"ls_clamp_mininterp"``,
        ``"hls_min_interp"``, ``"hls_max_interp"``, and ``"hls_full_interp"``.
        Please refer to the docstring of `Coulomb_logarithm` for more
        information about these methods.
```

**Review by namurphy — APPROVED** [review]
I have a few final minor formatting suggestions, but other than that I'd say it's ready.  @Tiger-Du — is there anything else you'd like to do with this before we merge it?  And thank you again!

**Comment by Tiger-Du** [comment]
Hi, Nick. Thank you for your helpful suggestions! I realize now that this PR must have been difficult to review since I made out-of-scope changes. (Sometimes I would get distracted and fix random things.) But yes, I think that this PR is just about done. One question about the unit tests: Should parametrization, rather than separate unit tests, be used for the abbreviations of the methods? @namurphy

**Comment by namurphy** [comment]
> One question about the unit tests: Should parametrization, rather than separate unit tests, be used for the abbreviations of the methods? @namurphy

It would be good to have the tests be parametrized, but I'd hold off on refactoring tests until #928 is done since that will add some test helper functionality.  I'll go ahead and merge this.  Thank you once again!

**Pull request merged**
